// flexi-select/esm2022/lib/flexi-select.module.mjs
import { NgModule } from '@angular/core';
import { FlexiSelectComponent } from './flexi-select.component';
import { FlexiOptionComponent } from './flexi-option.component';
import * as i0 from "@angular/core";
export class FlexiSelectModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.0.6", ngImport: i0, type: FlexiSelectModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "18.0.6", ngImport: i0, type: FlexiSelectModule, imports: [FlexiSelectComponent,
            FlexiOptionComponent], exports: [FlexiSelectComponent,
            FlexiOptionComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "18.0.6", ngImport: i0, type: FlexiSelectModule, imports: [FlexiSelectComponent] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.0.6", ngImport: i0, type: FlexiSelectModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [],
                    imports: [
                        FlexiSelectComponent,
                        FlexiOptionComponent
                    ],
                    exports: [
                        FlexiSelectComponent,
                        FlexiOptionComponent
                    ]
                }]
        }] });
