// flexi-select/esm2022/lib/flexi-option.component.mjs
import { Component, Input, ViewEncapsulation } from '@angular/core';
import * as i0 from "@angular/core";
export class FlexiOptionComponent {
    constructor(el) {
        this.el = el;
        this.viewValue = "";
    }
    ngOnInit() {
        this.viewValue = this.el.nativeElement.innerText.trim();
    }
    ngAfterViewInit() {
        this.viewValue = this.el.nativeElement.innerText.trim();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.0.6", ngImport: i0, type: FlexiOptionComponent, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "18.0.6", type: FlexiOptionComponent, isStandalone: true, selector: "flexi-option", inputs: { value: "value" }, ngImport: i0, template: '<ng-content></ng-content>', isInline: true, encapsulation: i0.ViewEncapsulation.None }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.0.6", ngImport: i0, type: FlexiOptionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'flexi-option',
                    standalone: true,
                    template: '<ng-content></ng-content>',
                    encapsulation: ViewEncapsulation.None
                }]
        }], ctorParameters: () => [{ type: i0.ElementRef }], propDecorators: { value: [{
                type: Input
            }] } });
