// flexi-grid/esm2022/lib/flexi-grid-column.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class FlexiGridColumnComponent {
    constructor() {
        this.field = '';
        this.title = '';
        this.sortable = true;
        this.filterable = true;
        this.visible = true;
        this.filterType = "text";
        this.format = null;
        this.fraction = 2;
        this.symbol = "";
        this.showSymbolInFront = true;
        this.width = "100%";
        this.resizable = true;
        this.draggable = true;
        this.className = "";
        this.textAlign = "left";
        this.hideOverflow = true;
        this.value = "";
        this.filterOperator = "contains";
    }
    ngOnChanges(changes) {
        if (changes['filterType']) {
            this.setDefaultFilterOperator();
        }
    }
    setDefaultFilterOperator() {
        if (this.filterType === 'number') {
            this.filterOperator = 'eq';
        }
        else {
            this.filterOperator = 'contains';
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.0.6", ngImport: i0, type: FlexiGridColumnComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "18.0.6", type: FlexiGridColumnComponent, selector: "flexi-grid-column", inputs: { field: "field", title: "title", sortable: "sortable", filterable: "filterable", visible: "visible", filterType: "filterType", columnTemplate: "columnTemplate", format: "format", fraction: "fraction", symbol: "symbol", showSymbolInFront: "showSymbolInFront", width: "width", resizable: "resizable", draggable: "draggable", className: "className", textAlign: "textAlign", hideOverflow: "hideOverflow" }, host: { properties: { "attr.host-id": "\"flexi-grid-column-1\"" } }, usesOnChanges: true, ngImport: i0, template: '', isInline: true }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.0.6", ngImport: i0, type: FlexiGridColumnComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'flexi-grid-column',
                    host: { '[attr.host-id]': '"flexi-grid-column-1"' },
                    template: ''
                }]
        }], propDecorators: { field: [{
                type: Input
            }], title: [{
                type: Input
            }], sortable: [{
                type: Input
            }], filterable: [{
                type: Input
            }], visible: [{
                type: Input
            }], filterType: [{
                type: Input
            }], columnTemplate: [{
                type: Input
            }], format: [{
                type: Input
            }], fraction: [{
                type: Input
            }], symbol: [{
                type: Input
            }], showSymbolInFront: [{
                type: Input
            }], width: [{
                type: Input
            }], resizable: [{
                type: Input
            }], draggable: [{
                type: Input
            }], className: [{
                type: Input
            }], textAlign: [{
                type: Input
            }], hideOverflow: [{
                type: Input
            }] } });
