// flexi-grid/esm2022/lib/flexi-grid.module.mjs
import { NgModule } from '@angular/core';
import { FlexiGridColumnComponent } from './flexi-grid-column.component';
import { FlexiGridComponent } from './flexi-grid.component';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { TrCurrencyPipe } from 'tr-currency';
import * as i0 from "@angular/core";
export class FlexiGridModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.0.6", ngImport: i0, type: FlexiGridModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "18.0.6", ngImport: i0, type: FlexiGridModule, declarations: [FlexiGridComponent,
            FlexiGridColumnComponent], imports: [CommonModule,
            FormsModule,
            TrCurrencyPipe], exports: [FlexiGridComponent,
            FlexiGridColumnComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "18.0.6", ngImport: i0, type: FlexiGridModule, imports: [CommonModule,
            FormsModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.0.6", ngImport: i0, type: FlexiGridModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        FlexiGridComponent,
                        FlexiGridColumnComponent
                    ],
                    imports: [
                        CommonModule,
                        FormsModule,
                        TrCurrencyPipe
                    ],
                    exports: [
                        FlexiGridComponent,
                        FlexiGridColumnComponent
                    ]
                }]
        }] });
